client/a8c-for-agencies/sections/signup/signup-form/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { Card } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { useCallback } from 'react';
+import AgencyDetailsForm from 'calypso/a8c-for-agencies/sections/signup/agency-details-form';
+import useCreateAgencyMutation from 'calypso/a8c-for-agencies/sections/signup/agency-details-form/hooks/use-create-agency-mutation';
+import AutomatticLogo from 'calypso/components/automattic-logo';
+import CardHeading from 'calypso/components/card-heading';
+import { useDispatch } from 'calypso/state';
+import { fetchAgencies } from 'calypso/state/a8c-for-agencies/agency/actions';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { errorNotice, removeNotice } from 'calypso/state/notices/actions';
+import type { AgencyDetailsPayload } from 'calypso/a8c-for-agencies/sections/signup/agency-details-form/types';
+import type { APIError } from 'calypso/state/a8c-for-agencies/types';
+
+import './style.scss';
+
+export default function SignupForm() {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const notificationId = 'a4a-agency-signup-form';
+
+	const createAgency = useCreateAgencyMutation( {
+		onSuccess: () => {
+			dispatch( fetchAgencies() );
+		},
+		onError: ( error: APIError ) => {
+			dispatch( errorNotice( error?.message, { id: notificationId } ) );
+		},
```

Comment:
Just realized this while I was trying to check this throughout the A4A codebase. 

onSuccess and onError are getting deprecated: https://tkdodo.eu/blog/breaking-react-querys-api-on-purpose

Reply:
Thanks for letting me know! Let's fix this in a follow-up PR.